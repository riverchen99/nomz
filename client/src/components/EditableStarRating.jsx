import React from 'react';
import StarRatingComponent from 'react-star-rating-component';

class EditableStarRating extends React.Component {
  constructor(props) {
    super(props);

    this.onStarClick = this.onStarClick.bind(this);
  }

  onStarClick(nextValue, prevValue, name) {
    console.log('name: %s, nextValue: %s, prevValue: %s', name, nextValue, prevValue);
    this.props.onRatingChange(nextValue);
  }

  render() {
    return (
      <div style={{fontSize: 50 }}>
        <StarRatingComponent 
          name="starRating"
          starColor="#FFC32B"
          emptyStarColor="#C4C4C4"
          value={this.props.star_rating}
          onStarClick={this.onStarClick} />
      </div>
    );
  }
}

export default EditableStarRating;
