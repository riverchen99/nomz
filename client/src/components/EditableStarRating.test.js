import React from 'react';
import { shallow, mount, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import EditableStarRating from './EditableStarRating';

configure({ adapter: new Adapter() });

const clickFn = jest.fn();
describe('Editable star rating component', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<EditableStarRating />);
    expect(component).toMatchSnapshot();
  });
  it('should render correctly with multiple props', () => {
    const component = mount(<EditableStarRating value={1} onStarClick={clickFn} />);
    expect(component.props().value).toEqual(1);
  });
  it('calls the onStarClick function correctly', () => {
    const component = mount(<EditableStarRating value={0} onStarClick={clickFn} />);
    component.props().onStarClick();
    expect(clickFn).toHaveBeenCalled();
  });
});
