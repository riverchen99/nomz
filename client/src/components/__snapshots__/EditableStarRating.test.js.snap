// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editable star rating component should render correctly with no props 1`] = `ShallowWrapper {}`;
